docs/source/user-guide/example-usage.md

```diff
@@ -81,6 +81,26 @@ async fn main() -> datafusion::error::Result<()> {
 +---+--------+
 ```
 
+## Arrow Versions
+
+Many of DataFusion's public APIs use types from the
```

Comment:
This is what I am trying to help

Reply:
This would be really helpful! Maybe also good to mention that it also applies to `parquet`.
It might be helpful to mention that this could also result in `ArrayRef` `downcast_ref` failure,
which just returns None and provides no error messages.

Reply:
I added a note in e420deb04 -- hopefully that makes senes